back/lib/tasks/data_migrate/20231127_update_idea_statuses_and_phases.rake

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# bin/rails "data_migrate:update_idea_statuses_and_phases['ideas.csv','participatie.leuven.be','nl-BE']"
+namespace :data_migrate do
+  desc 'Update Idea statuses and phases.'
+  task :update_idea_statuses_and_phases, %i[url host locale] => [:environment] do |_t, args|
+    data = CSV.parse(open(args[:url]).read, { headers: true, col_sep: ',', converters: [] })
+
+    Apartment::Tenant.switch(args[:host].tr('.', '_')) do
+      default_locale = args[:locale]
+      statuses = IdeaStatus.all.to_a
+      find_status_by_title = ->(title) { statuses.find { _1.title_multiloc[default_locale] == title } || raise("Idea status not found: '#{title}'.") }
+      create_status_by_title = lambda do |title, color|
+        IdeaStatus.find_or_create_by!(
+          title_multiloc: { default_locale => title },
+          description_multiloc: { default_locale => title },
+          color: color,
+          code: 'custom'
+        )
+      end
+
+      statuses_mapping = {
+        'verder onderzocht' => find_status_by_title.call('Wordt Onderzocht'),
+        'uitgevoerd' => find_status_by_title.call('uitgevoerd'),
+        'afgesloten/wordt niet verder onderzocht' => find_status_by_title.call('Afgesloten'),
+        'ingepland' => find_status_by_title.call('Ingepland'),
+        'uitvoering in 2024' => create_status_by_title.call('Uitvoering in 2023', '#216db7'),
+        'uitvoering in 2023' => create_status_by_title.call('Uitvoering in 2024', '#b721a7')
+      }
+
+      all_ideas = Idea.all.to_a
+
+      data.each do |d|
+        title = d['Titel'].strip
+        body = d['Beschrijving'].strip
+        body_first_word = body.split.first
+        ideas = all_ideas.select do |idea|
+          idea.title_multiloc.value?(title) &&
+            idea.body_multiloc.values.any? do |body_value|
+              sanitized = ActionView::Base.full_sanitizer.sanitize(body_value).strip
+              sanitized.starts_with?(body_first_word)
+            end
+        end
+        # Useful for testing:
+        # ideas = [Idea.new(title_multiloc: { default_locale => title }, body_multiloc: { default_locale => body }, project: Project.order(created_at: :asc).last, publication_status: 'published')]
+
+        if ideas.size != 1
+          puts "ERROR: Found #{ideas.size} ideas with title '#{title}' and body first word '#{body_first_word}'. Skipping update."
+          next
+        end
+
+        idea = ideas.first
+        target_phase = idea.project.phases.last
```

Comment:
Here again, I'd probably would have done this statically (e.g. as an arg to the rake task) since it should be the same for all. This works too of course, just more queries, and more chance of messing up data we rather not touch.